application/core/LSYii_Validators.php

```diff
@@ -184,4 +186,39 @@ public function multiLanguageFilter($value)
         $aValue = array_map("sanitize_languagecode", $aValue);
         return implode(" ", $aValue);
     }
+
+    /**
+     * Checks whether an URL seems unsafe in terms of XSS.
+     * @param string $url
+     * @return boolean Returns true if the URL is unsafe.
+     */
+    public static function isXssUrl($url)
+    {
+        $clean = self::removeInvisibleChars($url);
+
+        // Remove javascript:
+        if (stripos($clean, "javascript:") !== false) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Removes invisible characters from a string.
+     * @param string $string
+     * @return string
+     */
+    public static function removeInvisibleChars($string)
+    {
+        $clean = urldecode(html_entity_decode($string));
+
+        // Remove invisible characters
```

Comment:
newline and carriage return are invisble character ? Not space ?

Maybe removing space + invisble